Fix issue with in-place modification of FieldInfo (#4067)
samuelcolvin/pydantic: changes/4067-adriangb.md

> fix in-place modification of FieldInfo that made it impossible to use PEP 593 type aliases

Reviewer: ```suggestion
Fix in-place modification of `FieldInfo` that caused problems with PEP 593 type aliases
```
